pkg/reconciler/revision/resources/imagecache.go

```diff
@@ -27,15 +27,19 @@ import (
 )
 
 // MakeImageCache makes an caching.Image resources from a revision.
-func MakeImageCache(rev *v1.Revision) *caching.Image {
-	image := rev.Status.DeprecatedImageDigest
+func MakeImageCache(rev *v1.Revision, containerName, image string) *caching.Image {
+	// If the given image is empty then replace that with the image from spec.Container[*].Image.
 	if image == "" {
-		image = rev.Spec.GetContainer().Image
+		for _, v := range rev.Spec.Containers {
```

Comment:
might it be better to do this fall-back logic for ImageDigest not being there in the caller? that way we can avoid this loop

Reply:
Removed for loop completely
Its not required because in the called function we are passing the `Digest`  from the containerStatus so we can use directly the value coming from the called function